pandas/tseries/tests/test_tslib.py

```diff
@@ -519,7 +519,11 @@ def test_parsers(self):
                  '2014-06': datetime.datetime(2014, 6, 1),
                  '06-2014': datetime.datetime(2014, 6, 1),
                  '2014-6': datetime.datetime(2014, 6, 1),
-                 '6-2014': datetime.datetime(2014, 6, 1), }
+                 '6-2014': datetime.datetime(2014, 6, 1),
+
+                 '20010101 12': datetime.datetime(2001, 1, 1, 12),
+                 '20010101 1234': datetime.datetime(2001, 1, 1, 12, 34),
```

Comment:
hmm, I don't think ``12`` or ``1234`` should work (but ``123456``) should , IOW, you either have to fully specify HHMMSS or not at all. Any ambiuity in *partially* accepting it as you have here? 

cc @chris-b1